libraries/standard/mqtt/src/mqtt.c

```diff
@@ -326,33 +322,41 @@ static int32_t sendPacket( MQTTContext_t * pContext,
     sendTime = pContext->callbacks.getTime();
 
     /* Loop until the entire packet is sent. */
-    while( bytesRemaining > 0UL )
+    while( ( bytesRemaining > 0UL ) && ( sendError == false ) )
     {
         bytesSent = pContext->transportInterface.send( pContext->transportInterface.pNetworkContext,
                                                        pIndex,
                                                        bytesRemaining );
 
-        if( bytesSent > 0 )
+        if( bytesSent <= 0 )
```

Comment:
By first checking if unsigned bytesSent is less than zero, we avoid possible loss of information from casting bytesSent to unsigned in the second else case on line 337. 
By doing the error checks first we do same thing as checking the success condition (  ( bytesSend > 0 ) && (bytesSent <= bytesRemaining) ).